drivers/sensor/tsl2540/tsl2540.c

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright (c) 2022 T-Mobile USA, Inc.
+ * Copyright (c) 2023 T-Mobile USA, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ams_tsl2540
+
+#include "tsl2540.h"
+
+#include <stdlib.h>
+
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+
+LOG_MODULE_REGISTER(tsl2540, CONFIG_SENSOR_LOG_LEVEL);
+
+int tsl2540_reg_read(const struct device *dev, uint8_t reg, uint8_t *val)
+{
+	const struct tsl2540_config *cfg = dev->config;
+	int result;
+
+	result = i2c_reg_read_byte_dt(&cfg->i2c_spec, reg, val);
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+int tsl2540_reg_write(const struct device *dev, uint8_t reg, uint8_t val)
+{
+	const struct tsl2540_config *cfg = dev->config;
+	int result;
+
+	result = i2c_reg_write_byte_dt(&cfg->i2c_spec, reg, val);
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
```

Comment:
Wrappers like this are overkill unless the sensor supports multiple bus types (e.g., I2C and SPI). This sensor is I2C only, so please remove.